source/extensions/tracers/datadog/datadog_tracer_impl.cc

```diff
@@ -100,13 +100,19 @@ void TraceReporter::flushTraces() {
     ENVOY_LOG(debug, "submitting {} trace(s) to {} with payload size {}", pendingTraces,
               encoder_->path(), encoder_->payload().size());
 
-    Http::AsyncClient::Request* request =
-        driver_.clusterManager()
-            .httpAsyncClientForCluster(driver_.cluster()->name())
-            .send(std::move(message), *this,
+    if (driver_.clusterManager().get(driver_.cluster())) {
```

Comment:
Optimally you would use the thread local cluster manager callbacks here to avoid this lookup every time. Perhaps put in a TODO for that if you don't want to do it now? It's possible that a helper base/mix-in class could be used for this across tracers and other similar cases. (Comment applies to all the other tracers.)

Reply:
I've implemented it for `Zipkin` tracer. Please take a look.

Reply:
I've implemented cluster manager callbacks for the remaining tracers.